test/framework/workload.go

```diff
@@ -13,6 +13,7 @@ import (
 
 type WorkloadCluster struct {
 	*ClusterE2ETest
+	ManagementCluster               *ClusterE2ETest
```

Comment:
Is this necessary? I would prefer if the workload cluster access to the management cluster tests is very limited. From here the workload cluster code could do anything to the management cluster and that seems dangerous. We already have a very blurred line dividing responsibilities, this creates another big cross dependency.